nanoFramework.Tools.DebugLibrary.Shared/PortSerial/PortSerial.cs

```diff
@@ -69,75 +70,44 @@ public PortSerial(PortSerialManager portManager, NanoDevice<NanoSerialDevice> se
-            {
-                /////////////////////////////////////////////////////////////
-                // need to FORCE the parity setting to _NONE_ because        
-                // the default on the current ST Link is different causing 
-                // the communication to fail
-                /////////////////////////////////////////////////////////////
-
-                NanoDevice.DeviceBase = new SerialPort(InstanceId, BaudRate, Parity.None, 8);
-
-                if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
-                {
-                    Device.DtrEnable = true;
-                    Device.RtsEnable = true;
-                }
-
-                // Device could have been blocked by user or the device has already been opened by another app.
-                if (Device != null)
-                {
-                    try
-                    {
-                        Device.Open();
-                    }
-                    catch (IOException)
-                    {
-                        retry = true;
-                    }
-
-                    if (retry)
-                    {
-                        Thread.Sleep(100);
-                        Device.Open();
-                    }
-
-                    successfullyOpenedDevice = true;
-
-                    // set conservative timeouts
-                    Device.WriteTimeout = 5000;
-                    Device.ReadTimeout = 500;
-                    Device.ErrorReceived += Device_ErrorReceived;
-
-                    // better make sure the RX FIFO it's cleared
-                    Device.DiscardInBuffer();
-                }
-                else
-                {
-                    successfullyOpenedDevice = false;
-                }
-            }
-            catch(UnauthorizedAccessException)
-            {
-                return ConnectPortResult.Unauthorized;
-            }
-#if DEBUG
-            catch (Exception ex)
-#else
-            catch
-#endif
-            {
-                // catch all because the device open might fail for a number of reasons
-                return ConnectPortResult.ExceptionOccurred;
-            }
-
+
+            /////////////////////////////////////////////////////////////
+            // need to FORCE the parity setting to _NONE_ because        
+            // the default on the current ST Link is different causing 
+            // the communication to fail
+            /////////////////////////////////////////////////////////////
+
+            NanoDevice.DeviceBase = new SerialPort(InstanceId, BaudRate, Parity.None, 8);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                Device.DtrEnable = true;
+                Device.RtsEnable = true;
+            }
+
+            // Device could have been blocked by user or the device has already been opened by another app.
+            if (Device != null)
+            {
+                Device.Open();

```

Comment:
I don't see a try pattern around this open, if anything goes wrong, where is it catch?

Reply:
Yes Polly does the exception handling in the call. Polly is just the right tool for these kinds of problems.